app/user/layout.js

```diff
@@ -27,6 +30,15 @@ const options = [
 ]
 
 const Layout = ({ children }) => {
+  const { isLogin } = useShallowEqualSelector((state) => state.user)
+  const router = useRouter()
+
+  useComponentWillMount(() => {
```

Comment:
It is a bit weird, why don't you just use the useEffect instead of this one?

Reply:
> It is a bit weird, why don't you just use the useEffect instead of this one?

The `useEffect()` only works after the component is mounted. I created this `useComponentWillMount()` hook to perform secondary side-effects before the component is mounted